lib/providers/torrentleech.js

```diff
@@ -0,0 +1,88 @@
+const Promise = require('bluebird');
+const request = Promise.promisify(require('request'));
+const TorrentProvider = require('../TorrentProvider');
+
+class TorrentLeech extends TorrentProvider {
+	constructor() {
+		super({
+			name: 'TorrentLeech',
+			requireAuthentification: true,
+			supportCookiesAuthentification: true,
+			supportCredentialsAuthentification: false,
+			baseUrl: 'https://www.torrentleech.org',
+			searchUrl:
+				'/torrents/browse/list/query/{query}/categories/{cat}/orderby/seeders/order/desc',
+			categories: {
+				All:
+					'',
+				Movies: '8,9,11,37,43,14,12,13,47,15,29',
+				TV: '26,32,27',
+				Games: '17,42,18,19,40,20,21,39,22,28,30,48',
+				Apps: '23,24,25,33',
+				Education: '38',
+				Animation: '34,35',
+				Books: '45,46',
+				Music: '31,16',
+				Foreign: '36,44'
+			},
+			defaultCategory: 'All',
+		});
+	}
+
+	search(query, category) {
+		const url = this.getUrl(category, query);
+		if (url === null) {
+			return undefined;
+		}
+		return request({ url, jar: this.cookieJar, headers: { 'User-Agent': 'curl/7.37.0' } })
```

Comment:
is it possible to use "this.request" instead of "request" ?

Reply:
Yes, i will